src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function returnFunction(x) {
+    if (x === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += x;
+
+      return returnFunction;
+    }
```

Comment:
if x doesn't pass, the first condition converts to true and you return result. Everything that after return won’t run

```suggestion
      return result;
    }
    
    sum += x;

    return returnFunction;
```